lib/FrontendPlugin.cpp

```diff
@@ -191,7 +191,7 @@ struct ExtraValidationConsumer final : public ASTConsumer {
     }
 
     // First check if we have a pointer type.
-    if (Ty->isPointerType()) {
+    if (Ty->isPointerType() || (!IsKernelParameter && Ty->isReferenceType())) {
```

Comment:
Is `IsKernelParameter` required? Wouldn't kernel argument references be rejected elsewhere in clang?

Reply:
Doen't look like it: https://gcc.godbolt.org/z/fE84j8s7q